db/Pos/DB/Functions.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE AllowAmbiguousTypes       #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE ScopedTypeVariables       #-}
+
+-- | Basically wrappers over RocksDB library.
+
+module Pos.DB.Functions
+       ( openDB
+
+       -- * Key/Value helpers
+       , encodeWithKeyPrefix
+       , rocksDelete
+       , rocksGetBi
+       , rocksGetBytes
+       , rocksPutBi
+       , rocksPutBytes
+       , rocksDecodeWP
+       , rocksDecodeMaybe
+       , rocksDecodeMaybeWP
+       , rocksDecodeKeyValMaybe
+
+       -- * Batch
+       , RocksBatchOp (..)
+       , SomeBatchOp (..)
+       , SomePrettyBatchOp (..)
+       , rocksWriteBatch
+       ) where
+
+import qualified Data.ByteString       as BS (drop, isPrefixOf)
+import qualified Data.ByteString.Lazy  as BSL
+import           Data.Default          (def)
+import qualified Data.Text.Buildable
+import qualified Database.RocksDB      as Rocks
+import           Formatting            (bprint, sformat, shown, string, (%))
+import           Serokell.Util.Text    (listJson)
+import           Universum
+
+import           Pos.Binary.Class      (Bi, decodeFull, encodeStrict)
+import           Pos.DB.Error          (DBError (DBMalformed))
+import           Pos.DB.Iterator.Class (DBIteratorClass (..))
+import           Pos.DB.Types          (DB (..))
+
+openDB :: MonadIO m => FilePath -> m (DB)
+openDB fp = DB def def def
+                   <$> Rocks.open fp def
+                        { Rocks.createIfMissing = True
+                        , Rocks.compression     = Rocks.NoCompression
+                        }
+
+encodeWithKeyPrefix
+    :: forall i . (DBIteratorClass i, Bi (IterKey i))
+    => IterKey i -> ByteString
+encodeWithKeyPrefix = (iterKeyPrefix @i Proxy <>) . encodeStrict
```

Comment:
This is inefficient. It copies the whole bytestring in `encodeStrict`, and then appends `(<>)` which is `n` more steps. Slow stuff.